test/units/plugins/action/test_gather_facts.py

```diff
@@ -63,7 +63,7 @@ def test_network_gather_facts(self):
         self.assertEqual(mod_args['gather_subset'], 'min')
 
         facts_modules = C.config.get_config_value('FACTS_MODULES', variables=self.task_vars)
-        self.assertEqual(facts_modules, ['ansible.legacy.ios_facts'])
+        self.assertEqual(facts_modules, ['smart'])
```

Comment:
tests now make no sense since they also abused the modification of the setting, we should now check that when the option is 'smart' and network os is set we get the 'right' facts module

Reply:
The temporary list() patch + the test cleanup is being done in another PR to get module_defaults tests working (I marked this as a draft to implement the long-term solution). I improved the test coverage there, let me know if that's not what you were thinking https://github.com/ansible/ansible/pull/73864/commits/d19d8ec28664be6998c11c322ab686846b5b58fc